packages/lib-utils/src/components/table/Table.tsx

```diff
@@ -0,0 +1,95 @@
+// @ts-nocheck
+import * as React from 'react';
+import {
+  ICell,
+  OnSelect,
+  SortByDirection,
+  TableComposable,
+  TableGridBreakpoint,
+  Tbody,
+  // Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+import { StatusBox } from './StatusBox';
+
+export type RowProps<D, R extends any = {}> = {
+  obj: D;
+  rowData?: R; // ??! an object with row configuration eg { showNodes: .... }, added optionality for now
+  activeColumnIDs?: Set<string>; // ??! a set of columns visible in the table, added optionality for now..
+};
+
+export type TableColumn<D> = ICell & {
+  title: string;
+  id: string;
+  additional?: boolean;
+  // ??! sorting TO DO
+  sort?: ((data: D[], sortDirection: SortByDirection) => D[]) | string; // map to the new sort
+};
+
+type TableProps<D, R extends any = {}> = {
+  data: D[];
+  unfilteredData: D[];
+  loaded: boolean;
+  loadError: any;
```

Comment:
Since we have a chance to revisit our table component API, we should try avoiding `any` if possible